File: tree/count_leaf.py
class Node:
    def __init__ (self,key):
        self.left=None
        self.right=None
        self.data=key

def C(root):
    if root is None:
        return 0
    if root.left is None and root.right is None:
        return 1
    else:
        return C(root.left)+C(root.right)
        
        
root=Node(1)
root.left=Node(2)
root.right=Node(3)
root.left.right=Node(4)
root.right.right=Node(5)
root.right.left=Node(7)
print(C(root))

File: tree/count_leaf.cpp
#include<stdio.h>
#include<stdlib.h>
struct node
{
    int data;
    struct node *left,*right;
};
struct node* newnode(int data)
{
    struct node* n=(struct node*)malloc(sizeof(struct node));
    n->left=n->right=NULL;
    n->data=data;
    return n;
}
int countleaf(struct node* tr)
{
    if(tr==NULL)
        return 0;
    if(tr->left==NULL&&tr->right==NULL)
        return 1;
    else

        return(countleaf(tr->left)+countleaf(tr->right));
}
int main()
{
    struct node* tr=newnode(6);
    tr->left=newnode(3);
    tr->left->left=newnode(1);
    tr->left->right=newnode(5);
    tr->right=newnode(8);
    tr->left->right->right=newnode(30);
    tr->left->right->right->right=newnode(40);
    tr->right->right=newnode(45);
   printf(" no of leaf node.. %d",countleaf(tr));
    return 0;
}
